beacon-chain/core/blocks/signature.go

```diff
@@ -77,6 +78,27 @@ func VerifyBlockSignature(beaconState state.ReadOnlyBeaconState,
 	return helpers.VerifyBlockSigningRoot(proposerPubKey, sig, domain, rootFunc)
 }
 
+// VerifyBlockSignatureUsingCurrentFork verifies the proposer signature of a beacon block. This differs
+// from the above method by not using fork data from the state and instead retrieving it
+// via the respective epoch.
+func VerifyBlockSignatureUsingCurrentFork(beaconState state.ReadOnlyBeaconState, blk block.SignedBeaconBlock) error {
```

Comment:
Given that this is used in a sensitive path, thinking out loud here, can this method ever use a different fork object vs `Process_Slots(state).Fork` ? We should be sure on this before merging it in to `develop`.

Reply:
Should we just error if `blk.Slot() > beaconState.Slot()` ?